### ISSUE TITLE
Game crash on compile

### ISSUE THREAD

**Issue by untoldwind** [untoldwind/KontrolSystem2#51]
Kind of critical:

Game hangs up on compiling this:
```
use { Vessel } from ksp::vessel
use { CONSOLE } from ksp::console

const int_nil : Option<int> = None()
const int_one : Option<int> = Some(1)

const str_nil : Option<string> = None()
const str_one : Option<string> = Some("hello")

pub struct StructOne(p1: float, p2: string) {
    f1: float = p1
    f2: string = p2
}

const s1_none : Option<StructTwo> = None()
const s1_example : StructOne = StructOne(1.0, "foo")
const s1_some :  Option<StructTwo> = Some(s1_example)

// notice that StructTwo uses Option<StructOne> ... see below.

pub struct StructTwo(p1: float, p2: string, pn:  Option<StructTwo>) {
    f1: float = p1
    f2: string = p2
    next:  Option<StructTwo> = pn
}

const s2_none : Option<StructTwo> = None()
const s2_example : StructTwo = StructTwo(1.0, "foo", s2_none)
const s2_some : Option<StructTwo> = Some(s2_example)

pub fn main_flight( vessel: Vessel) -> Result<Unit, string> = {
    CONSOLE.print_line("hello from option.to2")
}
```

**Comment by untoldwind** [comment]
It is actually a stack-overflow: Type resolveing gets stuck in:
```
...
  at KontrolSystem.TO2.AST.OptionType.IsAssignableFrom (KontrolSystem.TO2.Generator.ModuleContext context, KontrolSystem.TO2.AST.TO2Type otherType) [0x00010] in <7347e8de3da34d0aae803f4143a677e4>:0
  at KontrolSystem.TO2.AST.RecordType.IsAssignableFrom (KontrolSystem.TO2.Generator.ModuleContext context, KontrolSystem.TO2.AST.TO2Type otherType) [0x00045] in <7347e8de3da34d0aae803f4143a677e4>:0
  at KontrolSystem.TO2.AST.OptionType.IsAssignableFrom (KontrolSystem.TO2.Generator.ModuleContext context, KontrolSystem.TO2.AST.TO2Type otherType) [0x00010] in <7347e8de3da34d0aae803f4143a677e4>:0
  at KontrolSystem.TO2.AST.RecordType.IsAssignableFrom (KontrolSystem.TO2.Generator.ModuleContext context, KontrolSystem.TO2.AST.TO2Type otherType) [0x00045] in <7347e8de3da34d0aae803f4143a677e4>:0
  at KontrolSystem.TO2.AST.OptionType.IsAssignableFrom (KontrolSystem.TO2.Generator.ModuleContext context, KontrolSystem.TO2.AST.TO2Type otherType) [0x00010] in <7347e8de3da34d0aae803f4143a677e4>:0
...
```

**Comment by untoldwind** [comment]
Should be now fixed in 0.2.1

**Comment by untoldwind** [comment]
Closing this for now as it does not have reappeared yet
